client/changelist/files_changelist.go

```diff
@@ -0,0 +1,103 @@
+package changelist
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"time"
+
+	"code.google.com/p/go-uuid/uuid"
+)
+
+type fileChangelist struct {
+	dir string
+}
+
+func NewFileChangelist(dir string) (*fileChangelist, error) {
+	err := os.MkdirAll(dir, 0600)
+	if err != nil {
+		return nil, err
+	}
+	return &fileChangelist{dir: dir}, nil
+}
+
+func (cl fileChangelist) List() []Change {
+	changes := make([]Change, 0)
+	dir, err := os.Open(cl.dir)
+	if err != nil {
+		return changes
```

Comment:
If the directory doesn't exist, I felt returning an empty list was appropriate. No dir, and empty dir, are functionally the same thing for us.